### PR TITLE
feat: RDF 1.2 reifier migration + Prolog→SHACL + addListener→subscribeQuery + N+1 elimination

### PR DESCRIPTION
**Pull request by HexaField**
## Summary

This PR combines the full SPARQL 1.2 migration for Flux — originally split across `feat/sparql-1.2` (#590, now closed) and `feat/sparql-1.2-cleanup` (#591). Together they deliver: all custom SPARQL queries migrated to RDF 1.2 reifier patterns, all Prolog replaced with SHACL APIs, all global `addListener` patterns replaced with targeted `subscribeQuery`, and all N+1 query patterns eliminated.

**Companion:** AD4M [coasys/ad4m#803](https://github.com/coasys/ad4m/pull/803)

---

## Part 1: Migrate Flux SPARQL Queries to RDF 1.2 Reifier Patterns

*(Originally #590)*

Migrates all custom SPARQL queries in Flux API packages from the old named-graph `GRAPH ?g { s p o }` pattern to direct default-graph triples with RDF 1.2 reifier patterns for link metadata.

#### `packages/api/src/channel/index.ts`
- `allItems()` — direct triple pattern, no GRAPH
- `unprocessedItems()` — 3 subqueries converted (all items, processed items, set-difference)
- `totalItemCount()` — direct pattern
- `recentConversations()` — reifier for timestamp
- `pinnedConversations()` — direct pattern

#### `packages/api/src/conversation/index.ts`
- `stats()` subgroups query — direct pattern
- `topics()` — direct pattern (5 GRAPH vars → direct triples)
- `subgroupsData()` main query — reifier for timestamp on `has_child` link
- `subgroupsData()` batch timestamp query — reifier for channel timestamp

#### `packages/api/src/conversation-subgroup/index.ts`
- `stats()` — direct pattern
- `topics()` — direct pattern
- `itemsData()` — reifier for both subgroup-item timestamp and type-link author, plus channel timestamp
- `topicsWithRelevance()` — direct pattern

#### `app/src/utils/registerMobileNotifications.ts`
- Mobile push notification trigger query — removed GRAPH wrapper

#### Other
- Added type-checking support in the CI pipeline
- Improved timeline subscription refresh for more responsive conversation updates
- Safer handling of missing channel data to prevent navigation/display errors
- Normalized community IDs to fix membership edge cases
- Improved mobile notification matching and AI task processing reliability
- More accurate timestamp selection for conversation ordering
- Expanded unit and integration tests for channels, conversations, and timeline subscriptions
- Integrated linting and type-checking gates into CI; updated ESLint config and ignore list

### Pattern Migration

**Before (named graphs):**
```sparql
GRAPH ?link1 { <parent> <ad4m://has_child> ?id . }
?link1 <ad4m://ontology/timestamp> ?timestamp .
```

**After (RDF 1.2 reifiers):**
```sparql
PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
<parent> <ad4m://has_child> ?id .
?_reifier rdf:reifies <<( <parent> <ad4m://has_child> ?id )>> .
?_reifier <ad4m://ontology/timestamp> ?timestamp .
```

---

## Part 2: Prolog → SHACL, addListener → subscribeQuery, N+1 Elimination

*(Originally #591)*

### 26 files changed, 514 insertions, 659 deletions (net −145 lines)

After this part, zero `perspective.infer()` calls, zero unfocused `addListener` patterns, and zero global link bus usage remain in Flux app/view source code.

### Prolog → SPARQL Conversions

| File | What changed |
|------|-------------|
| `topic/index.ts` | `linkedConversations()` / `linkedSubgroups()` → SPARQL |
| `semantic-relationship/index.ts` | All 5 methods → SPARQL |
| `conversation/util.ts` | `findEmbeddingSRId()` → SPARQL |

### `addListener` → Targeted `subscribeQuery`

| File | What changed |
|------|-------------|
| `flux-container.ts` | Global listener → `subscribeQuery` with Channel SPARQL |
| `CommentSection.tsx` | Global listener → `subscribeQuery` on source |
| `CommunityGraph.tsx` | Global listener → `subscribeQuery` on source/target (was leaking, now properly cleaned up) |
| `useAssociations.tsx` | Global listener → `subscribeQuery` with proper dispose |
| `TimelineColumn.tsx` | Global listener → `subscribeQuery` |
| `useCommunityService.ts` | Participant tracking → `subscribeQuery` on CHANNEL predicate |
| `cache.tsx` | Deleted unused `subscribeToPerspective` / `unsubscribeToPerspective` (zero callers, contained `addListener`) |

### N+1 Query Elimination

| File | What changed |
|------|-------------|
| `useCommunityService.ts` | Nested `Promise.all` → two-phase batch (link queries then conversation lookups) |
| `CommentSection.tsx` | N × `Message.get()` → `Message.findAll({ parent })` |
| `PollCard.tsx` | Per-answer `Vote.findAll` → `useLiveQuery(Answer, { include: { votes: true } })` |

### kanban-view Migration

| File | What changed |
|------|-------------|
| `Board.tsx` | `.infer()` → `listRegisteredClasses()` + `getNamedOptions()` + `addNamedOption()` + `ensureSDNASubjectClass(Task)`. Task creation uses `initialValues`. |
| `Entry.tsx` | `.infer(property_named_option)` → `getNamedOptions()` |
| `TaskModel.ts` | **New** — TypeScript `@Model` class with `@Property({ options: [...] })` for status enum |
| `Task.pl` | **Deleted** (22 lines of Prolog SDNA) |

### table-view Migration

| File | What changed |
|------|-------------|
| `TableView.tsx` | `.infer(subject_class)` → `listRegisteredClasses()` |
| `Table.tsx` | `.infer(property_named_option)` → `getNamedOptions()`. Updates use `getClassShape()` for predicate resolution + direct link manipulation. |
| `Header.tsx` | `.infer()` → `getInstanceClasses()` + `getSubjectData()`. Updates via `getClassShape()` + link add/remove. |
| `Entry.tsx` | Same as Header |
| `History.tsx` | Same as Header (read-only) |
| `NewClass.tsx` | `generateSDNA()` (raw Prolog) → `buildSHACLShape()` + `addShacl()`. Removed `makeRandomPrologAtom()`. |

### Global Link Bus Removal

| File | What changed |
|------|-------------|
| `MainView.vue` | Removed `gotNewMessage`, `onLinkAdded`, and associated `addLink` bus emissions |
| `usePerspectives.ts` | **Deleted** `addListeners()`, `onLinkAdded()`, `onLinkRemoved()`, and link callback arrays — the global link event bus is gone |

### Other

- `createCommunity.ts`: `.timestamp` → `.createdAt` (deprecated alias removed in companion PR)

---

## Depends On

AD4M [coasys/ad4m#803](https://github.com/coasys/ad4m/pull/803) — provides Oxigraph 0.5.7 + RDF 1.2 reifiers, Rust model query engine, `getNamedOptions()`, `addNamedOption()`, `listRegisteredClasses()`, `getClassShape()`, `getInstanceClasses()`, `getSubjectData()`, `SHACLShape`, `@Property({ options })`, `perspectiveModelSubscribe`

---

## Expected Performance Impact

- **Link listener overhead**: Near-zero — global `addListener` callbacks (fired on every link change across every perspective) replaced with targeted `subscribeQuery` that only fires when matching predicates change. In a community with 10+ channels, this eliminates ~90% of spurious JS callback invocations.
- **N+1 queries**: Batch operations (`findAll` with `include`) replace per-item `.get()` calls, reducing round-trips from N+1 to 1-2 per list render.
- **kanban/table views**: Prolog `infer()` calls (which loaded and evaluated the entire SDNA program) replaced with direct SHACL API lookups.
- **Vue reactivity**: Removing the global link bus prevents unnecessary re-renders in components that don't care about the changed links.

## Summary by CodeRabbit

* **New Features**
  * Lightweight channel and message summary models for faster list rendering.

* **Improvements**
  * More reliable live updates via targeted SPARQL subscriptions (faster, more responsive timelines and graphs).
  * Better community/channel ID handling (accepts private:// and shared formats).
  * Query and data-loading optimizations for quicker channel/conversation views and more robust AI task processing.

* **Tests**
  * Test suite migrated to Vitest with expanded coverage for channels, conversations, embeddings, and processing logic.

* **Chores**
  * CI expanded to run lint, build, and informational typecheck; added project-level typecheck script and task; dependency/version bumps for AD4M-related packages.

### CODE REVIEW

File: app/src/components/conversation/timeline/__tests__/timeline-subscription.test.ts [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+
+describe('AI processing trigger chain', () => {
+  it('checkIfWeShouldProcessTask requires MIN_ITEMS + DELAY items', () => {
+    const MIN_ITEMS_TO_PROCESS = 5;
+    const PROCESSING_ITEMS_DELAY = 3;
+    const threshold = MIN_ITEMS_TO_PROCESS + PROCESSING_ITEMS_DELAY;
+
+    expect(7).toBeLessThan(threshold);
+    expect(threshold).toBe(8);
+    expect(9).toBeGreaterThan(threshold);
```
Comment (lucksus) [inline app/src/components/conversation/timeline/__tests__/timeline-subscription.test.ts]: WTF?! :D

**Review by lucksus — COMMENTED** [review]
I'm a bit weary about those new tests. For my changes of the conversation query in channel.ts I had to adapt the test because it was testing the internal implementation of that function and not its performance. Looking at some of the other tests (have not done a full review) I see some nonsensical tests that test if JS can do arithmetic. And more tests that tests the inner implementation of functions but not what they do. Can't say if that's the case for all added tests but most look they only give us a false sense of safety but actually only make it harder to change code - without providing much benefit.
